Maybe 10

Klitsenko opened the pull request.

Klitsenko committed:
Merge remote-tracking branch 'origin/dev' into MAYBE-10

Klitsenko committed:
Task 10 completed

Klitsenko committed:
Merge remote-tracking branch 'origin/dev' into MAYBE-10

# Conflicts:
#	pom.xml
#	src/main/java/com/maybe/maybe/entity/Order.java
#	src/main/java/com/maybe/maybe/exception/RestExceptionHandler.java
#	src/main/java/com/maybe/maybe/repository/CustomerRepository.java
#	src/main/java/com/maybe/maybe/repository/DeskRepository.java
#	src/main/java/com/maybe/maybe/service/DeskService.java
#	src/main/resources/application.properties
#	src/main/resources/db/data/desk.csv

Klitsenko committed:
MAYBE 10 completed

deivur commented on src/main/java/com/maybe/maybe/exception/RestExceptionHandler.java:
```diff
@@ -46,6 +47,13 @@
 
+    @ExceptionHandler(BeanCreationException.class)
+    protected ResponseEntity<Object> handleEnityCreation(
+            BeanCreationException ex) {
```
Что это такое и зачем нужно?

deivur commented on src/main/java/com/maybe/maybe/service/DeskService.java:
```diff
@@ -13,7 +18,50 @@ public DeskService(DeskRepository deskRepository) {
+        if (!deskRepository.findAll().isEmpty()) {
+           return deskRepository.findAll();
+        } else {
+            throw new EntityNotFoundException("Can not find any desk");
```
Считаю, что тут не должно быть ексепшена, а должен просто возвращаться пустой массив.

deivur commented on src/main/java/com/maybe/maybe/service/DeskService.java:
```diff
@@ -13,7 +18,50 @@ public DeskService(DeskRepository deskRepository) {
+            desk.setName(deskDTO.getName());
+            return deskRepository.save(desk);
+        } else {
+            throw new BeanCreationException("Can not create desk entity by dto " + deskDTO.toString());
```
Считаю в корне неправильным! То что в ДТО обьекте нет нейма - это ошибка валидации, по мимо всего прочего ентити не являются спринговыми бинами.

deivur commented on src/main/java/com/maybe/maybe/service/DeskService.java:
```diff
@@ -13,7 +18,50 @@ public DeskService(DeskRepository deskRepository) {
+
+    public Desk updateById(Long id, DeskDTO deskDTO){
+        if(deskRepository.findDeskById(id) != null){
+            Desk desk = new Desk();
```
При апдейте не должен использоваться оператор new - нужно находить существующую сущность и апдейтить ее.

deivur commented on src/main/java/com/maybe/maybe/service/EmployeeService.java:
```diff
@@ -15,7 +20,57 @@ public EmployeeService(EmployeeRepository employeeRepository) {
+        if (!employeeRepository.findAll().isEmpty()) {
+            return employeeRepository.findAll();
+        } else {
+            throw new EntityNotFoundException("Can not find any employee");
```
Не должно быть ексепшена - а только пустой лист.

deivur commented on src/main/java/com/maybe/maybe/service/EmployeeService.java:
```diff
@@ -15,7 +20,57 @@ public EmployeeService(EmployeeRepository employeeRepository) {
+    }
+
+    public Employee createFromDTO(EmployeeDTO employeeDTO) {
+        if (employeeRepository.findEmployeeByLogin(employeeDTO.getLogin()) == null) {
```
Опять же ошибка валидации, не имеющая отношения к бизнес логике.

deivur commented on src/main/resources/application.properties:
```diff
@@ -1,8 +1,8 @@
 spring.liquibase.change-log=classpath:/db/changelog/changelog-master.xml
 spring.datasource.driver-class-name=org.postgresql.Driver
-spring.datasource.url=jdbc:postgresql://localhost:5432/postgres
+spring.datasource.url=jdbc:postgresql://localhost:32773/postgres
```
Не нужно всем менять на твои конфиги.

Klitsenko committed:
Desk service
fix: service\DeskService.java: change new Entity bu exist entity in updateById()

Klitsenko committed:
Merge branch 'dev' into MAYBE-10

Klitsenko committed:
Merge remote-tracking branch 'origin/dev' into MAYBE-10

# Conflicts:
#	pom.xml
#	src/main/java/com/maybe/maybe/controller/DeskController.java
#	src/main/java/com/maybe/maybe/entity/Desk.java
#	src/main/java/com/maybe/maybe/repository/CustomerRepository.java
#	src/main/java/com/maybe/maybe/repository/DeskRepository.java
#	src/main/java/com/maybe/maybe/service/DeskService.java
#	src/main/java/com/maybe/maybe/service/EmployeeService.java
#	src/main/resources/application.properties
#	src/main/resources/db/data/desk.csv

Klitsenko committed:
Desk controller
fix: controller\DeskController.java: resolve conflict

Klitsenko committed:
Desk service
fix: service\DeskService.java: resolve conflict

Klitsenko committed:
Merge remote-tracking branch 'origin/MAYBE-10' into MAYBE-10

# Conflicts:
#	pom.xml
#	src/main/java/com/maybe/maybe/controller/DeskController.java
#	src/main/java/com/maybe/maybe/repository/DeskRepository.java
#	src/main/java/com/maybe/maybe/service/DeskService.java
#	src/main/resources/application.properties
#	src/main/resources/db/data/desk.csv

Klitsenko committed:
Delete
refactor: dto\DeskRequest.java: delete unnecessary class
refactor: dto\EmployeeRequest.java: delete unnecessary class

Klitsenko committed:
Pom
fix: pom.xml: delete scope

Klitsenko committed:
Invoice service
fix: service\InvoiceService.java: change method name to findById
